zio-sbt-ecosystem/src/main/scala/zio/sbt/ZioEcosystemProjectPlugin.scala

```diff
@@ -0,0 +1,194 @@
+
+import scala.collection.immutable.ListMap
+
+import _root_.scalafix.sbt.ScalafixPlugin
+import org.scalafmt.sbt.ScalafmtPlugin
+import sbt.Keys._
+import sbt._
+import sbtbuildinfo.BuildInfoPlugin
+import sbtbuildinfo.BuildInfoPlugin.autoImport._
+import scalafix.sbt.ScalafixPlugin.autoImport._
+
+object ZioEcosystemProjectPlugin extends AutoPlugin {
+
+  override def trigger = allRequirements
+
+  override def requires: Plugins =
+    super.requires && de.heikoseeberger.sbtheader.HeaderPlugin && ScalafixPlugin && ScalafmtPlugin && BuildInfoPlugin
+
+  object autoImport extends Versions {
+
+    sealed trait ZIOSeries {
+      def version: String
+    }
+
+    object ZIOSeries {
+
+      case object Series1X extends ZIOSeries {
+        override val version: String = versions.zio1xVersion
+      }
+
+      case object Series2X extends ZIOSeries {
+        override val version: String = versions.zio2xVersion
+      }
+
+    }
+
+    def buildInfoSettings(packageName: String): Seq[Setting[_ <: Object]] =
+      Seq(
+        buildInfoKeys := Seq[BuildInfoKey](
+          name,
+          version,
+          scalaVersion,
+          sbtVersion,
+          isSnapshot
+        ),
+        buildInfoPackage := packageName
+      )
+
+    def addCommand(commandString: List[String], name: String, description: String): Seq[Setting[_]] = {
+      val cCommand = Commands.ComposableCommand(commandString, name, description)
+      Seq(
+        commands += cCommand.toCommand,
+        usefulTasksAndSettings += cCommand.toItem
+      )
+    }
+
+    val zioSeries: SettingKey[ZIOSeries] = settingKey[ZIOSeries]("Indicates whether to use ZIO 2.x or ZIO 1.x.")
+
+    val needsZio: SettingKey[Boolean] = settingKey[Boolean]("Indicates whether or not the project needs ZIO libraries.")
+
+    val welcomeBannerEnabled: SettingKey[Boolean] =
+      settingKey[Boolean]("Indicates whether or not to enable the welcome banner.")
+
+    val usefulTasksAndSettings: SettingKey[Map[String, String]] = settingKey[Map[String, String]](
+      "A map of useful tasks and settings that will be displayed as part of the welcome banner."
+    )
+
+  }
+
+  // Keep this consistent with the version in .core-tests/shared/src/test/scala/REPLSpec.scala
+  makeReplSettings {
+    """|import zio._
+       |import zio.console._
+       |import zio.duration._
+       |import zio.Runtime.default._
+       |implicit class RunSyntax[A](io: ZIO[ZEnv, Any, A]){ def unsafeRun: A = Runtime.default.unsafeRun(io.provideLayer(ZEnv.live)) }
+    """.stripMargin
+  }
+
+  // Keep this consistent with the version in .streams-tests/shared/src/test/scala/StreamREPLSpec.scala
+  makeReplSettings {
+    """|import zio._
+       |import zio.console._
+       |import zio.duration._
+       |import zio.stream._
+       |import zio.Runtime.default._
+       |implicit class RunSyntax[A](io: ZIO[ZEnv, Any, A]){ def unsafeRun: A = Runtime.default.unsafeRun(io.provideLayer(ZEnv.live)) }
+    """.stripMargin
+  }
```

Comment:
Just a question, how do you use these two settings?

Reply:
I think this might just be dead code I had pulled over from the random assortment `BuildHelpers` I sourced the plugin from.

My process was to port everything over into the plugin and cull anything that was never used.